### ISSUE TITLE
Question: is it possible to make snippets work only on beginning of the line?

### ISSUE THREAD

**Issue by cloggier** [hrsh7th/vim-vsnip#155]
Ultisnips [supports](https://github.com/SirVer/ultisnips/blob/c270950492d71bac0317d47d42cd0884eefe6490/doc/UltiSnips.txt#L649) it.
I am not sure if such option is part of VSCode snippets' format (couldn't find anything on that in documentation linked in this repo's README) or whether vim-vsnip does/could do itty itself.

Thanks!

**Comment by hrsh7th** [comment]
Currently, No.

It maybe related to https://github.com/hrsh7th/vim-vsnip/issues/136

**Comment by ghyatzo** [comment]
No, you probably can get the desired behaviour by implementing a custom vim function and then call it through the `VIM` syntax, e.g.
`${VIM: (col('.') == 1) ? 'snippet' : 'prefix'}`

It always expands, but if not at the beginning of the line it behaves like a noop. It is hacky and does not support tabstops or similar behaviours though. :D

**Comment by cloggier** [comment]
Don't like the hackie idea, but thanks! Seems like `line_pattern` from #136 might do that? Will wait for it. Thanks for great plugin!
